src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,43 @@
 package core.basesyntax;
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        File firstFile = new File(fromFileName);
+        File secondFile = new File(toFileName);
+
+        int buy = 0;
+        int supply = 0;
+        int result = 0;
+
+        try {
+            List<String> stringList = Files.readAllLines(firstFile.toPath());
+            for (String s : stringList) {
+                if (s.contains("buy")) {
+                    buy += Integer.parseInt(s.substring(s.indexOf(',') + 1, s.length()));
+                }
+                if (s.contains("supply")) {
+                    supply += Integer.parseInt(s.substring(s.indexOf(',') + 1, s.length()));
+                }
+            }
+
+            result = supply - buy;
+
+            StringBuilder stringBuilder = new StringBuilder();
+            stringBuilder
+                    .append("supply,").append(supply).append(System.lineSeparator())
+                    .append("buy,").append(buy).append(System.lineSeparator())
+                    .append("result,").append(result).append(System.lineSeparator());
 
+            secondFile.createNewFile();
+            Files.write(secondFile.toPath(), stringBuilder.toString().getBytes());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
You should provide a more detailed message when throwing a RuntimeException. It's not a good practice to throw an exception without any message because it makes debugging more difficult. The message should give an idea about what went wrong. For example, you could say 'Can't read data from the file ' + fileName.